[JIT] remove tuple logic in constant propagation

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31841 [JIT] implement tuple constants
* **#31840 [JIT] remove tuple logic in constant propagation**
* #31839 [JIT] improve mayContainAlias
* #31791 Add str[] float[] constants resubmit

The next PR in this stack makes tuples insertable as constants, so we can remove special handling of tuples in constant propagation.

Differential Revision: [D19439515](https://our.internmc.facebook.com/intern/diff/D19439515)

Commit:
remove tuple logic in constant propagation

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit baa78e4b:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=baa78e4bb642d04092fb91af63079510ca2527b0).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4120600) pytorch_linux_xenial_py2_7_9_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=79927817) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=79927817))
<details>
<summary>
<code>Jan 03 20:52:21 RuntimeError: test_jit failed!</code>
</summary>

```
Jan 03 20:52:21 Ran 2313 tests in 74.827s 
Jan 03 20:52:21  
Jan 03 20:52:21 FAILED (errors=1, skipped=81, expected failures=1) 
Jan 03 20:52:21  
Jan 03 20:52:21 Generating XML reports... 
Jan 03 20:52:21 Traceback (most recent call last): 
Jan 03 20:52:21   File "test/run_test.py", line 456, in <module> 
Jan 03 20:52:21     main() 
Jan 03 20:52:21   File "test/run_test.py", line 449, in main 
Jan 03 20:52:21     raise RuntimeError(message) 
Jan 03 20:52:21 RuntimeError: test_jit failed! 
Jan 03 20:52:21 + cleanup 
Jan 03 20:52:21 + retcode=1 
Jan 03 20:52:21 + set +x 
Jan 03 20:52:21 =================== sccache compilation log =================== 
Jan 03 20:52:22 ERROR:sccache::server: Compilation failed: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "/tmp/torch_extensions/test_compilation_error_formatting/main.cpp: In function \'int main()\':\n/tmp/torch_extensions/test_compilation_error_formatting/main.cpp:2:23: error: expected \';\' before \'}\' token\n int main() { return 0 }\n                       ^\n" } 
Jan 03 20:52:22  
Jan 03 20:52:22 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 03 20:52:22 Compile requests                 51 
Jan 03 20:52:22 Compile requests executed        27 
Jan 03 20:52:22 Cache hits                       20 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

suo commented:
I think I'm missing some context here—can we update the PR desc with some info about why this change is happening?

eellison commented:
@suo updated

Commit:
Update on "[JIT] remove tuple logic in constant propagation"


The next PR in this stack makes tuples insertable as constants, so we can remove special handling of tuples in constant propagation.


[ghstack-poisoned]

driazati commented on torch/csrc/jit/passes/constant_propagation.cpp:
```diff
@@ -117,34 +101,6 @@ struct ConstantPropagator {
     return var_outputs;
   }
 
-  // Tuples are not representable as constants, however
```
This sounds like since tuples are representable as constants they wouldn't need to be handled here, but that doesn't happen until the PR above this one in the stack, so how does this still work?

eellison replied:
i'm going to land the stack in one.

Commit:
Update on "[JIT] remove tuple logic in constant propagation"


The next PR in this stack makes tuples insertable as constants, so we can remove special handling of tuples in constant propagation.


[ghstack-poisoned]

Commit:
Update on "[JIT] remove tuple logic in constant propagation"


The next PR in this stack makes tuples insertable as constants, so we can remove special handling of tuples in constant propagation.


[ghstack-poisoned]

Commit:
Update on "[JIT] remove tuple logic in constant propagation"


The next PR in this stack makes tuples insertable as constants, so we can remove special handling of tuples in constant propagation.

Differential Revision: [D19439515](https://our.internmc.facebook.com/intern/diff/D19439515)

[ghstack-poisoned]
